### PR TITLE
Added live events support for attachments + wiki pages fix

### PR DESCRIPTION
**Pull request by simong**
This PR brings in support for creating, updating and removing attachments.

It also fixes an issue when deleting a wiki page wouldn't trigger a live event

Test plan:
  * create attachments by uploading course files
  * update display name of course files
  * delete some course files
  * delete a course wiki page
  * ensure appropriate messages are in the kinesis queue

### CODE REVIEW

File: app/observers/live_events_observer.rb [hunk 1]
```diff
@@ -19,9 +20,19 @@ def after_update(obj)
...
+        # Wiki pages are soft deleted rather than destroyed
+        Canvas::LiveEvents.wiki_page_deleted(obj)
       end
     when Assignment
       Canvas::LiveEvents.assignment_updated(obj)
+    when Attachment
+      if obj.display_name_changed?
+        Canvas::LiveEvents.attachment_updated(obj, obj.display_name_was)
+      elsif obj.file_state_changed? && obj.file_state == 'deleted'
+        # Attachments are soft deleted rather than destroyed
```
Comment (zwily) [inline app/observers/live_events_observer.rb]: Are they *always* soft deleted?

Comment (simong) [inline app/observers/live_events_observer.rb]: Nearly always AFAICT, but there were a few spots where `destroy_permanently` is called, so I added the `after_destroy` trigger back in.

**Comment by simong** [comment]
I also added some more properties to the attachment event.

**Comment by zwily** [comment]
@simong - Would you mind rebasing this onto the master branch and re-pushing? It looks like there are a few conflicts.

**Comment by simong** [comment]
I've rebased this against master, but the PR is still against `stable`. I'll close this and open a new one against master

**Comment by simong** [comment]
Opened a new PR against master at https://github.com/instructure/canvas-lms/pull/882

**Pull request closed without merging**